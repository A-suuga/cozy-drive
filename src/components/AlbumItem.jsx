import styles from '../styles/albumsList'

import React, { Component } from 'react'
// import classNames from 'classnames'
import { Link, withRouter } from 'react-router'
import { translate } from '../lib/I18n'

import { getPhotoLink } from '../actions/photos'

export class AlbumItem extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isLoading: true
    }

    if (props.album && props.album.photosIds.length) {
      getPhotoLink(props.album.photosIds[0])
        .then(link => this.setState({
          url: link,
          isLoading: false
        }))
    } else {
      this.setState({
        url: '',
        isLoading: false
      })
    }
  }

  render () {
    const { t, album, router } = this.props
    const { isLoading, url } = this.state
    const parentPath = router.location.pathname
    return (
      !isLoading &&
        <div className={styles['pho-album']}>
          <Link
            to={`${parentPath}/${album._id}`}
            className={styles['pho-album-link']}
          >
            <img
              className={styles['pho-album-photo-item']}
              src={url}
            />
            <h2 className={styles['pho-album-title']}>{album.name}</h2>
            <h4 className={styles['pho-album-description']}>
              {t('Albums.album_item_description',
                {smart_count: album.photosIds.length})
              }
            </h4>
          </Link>
        </div>
    )
  }
}

export default translate()(withRouter(AlbumItem))
