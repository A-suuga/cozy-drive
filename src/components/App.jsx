import styles from '../styles/app'

import React from 'react'
import { translate } from '../lib/I18n'
import classNames from 'classnames'

import Sidebar from './Sidebar'

const App = ({ t, children }) => (
  <div className={classNames(styles['pho-wrapper'], styles['coz-sticky'])}>

    <Sidebar />

    <main className={styles['pho-content']}>
      { children }
    </main>
  </div>
)

export default translate()(App)
