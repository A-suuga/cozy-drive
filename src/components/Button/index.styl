@require 'cozy-ui'

@media (max-width: (1023/basefont)rem)
    .dri-btn--more
        display inline-flex

@media (max-width: (768/basefont)rem)
    .dri-btn--more
        border        0
        margin-right  0
        height        em(48px, 14px)
        opacity       .5