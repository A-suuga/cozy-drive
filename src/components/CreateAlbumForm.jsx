import styles from '../styles/createAlbumForm'

import React, { Component } from 'react'
import { translate } from '../lib/I18n'

import classNames from 'classNames'

class CreateAlbumForm extends Component {
  constructor () {
    super()
    this.reset()
  }

  reset () {
    this.setState({
      name: '',
      isSubmitDisabled: true,
      isBusy: false,
      hasError: false
    })
  }

  onSubmit (event, callback) {
    event.preventDefault()
    this.setState({ isBusy: true })
    return callback(this.state.name)
      .then(
        () => this.reset(),
        () => this.setState({
          hasError: true,
          isBusy: false
        })
      )
  }

  onNameChange (event) {
    const name = event.target.value.trim()
    this.setState({
      name: name,
      isSubmitDisabled: !name.length,
      hasError: false
    })
  }

  render ({ t, onSubmitNewAlbum }) {
    return (
      <form onSubmit={(event) => this.onSubmit(event, onSubmitNewAlbum)}>
        <label className={styles['coz-create-album-label']}>
          {t('Albums.create.inline_form.create_label')}
        </label>
        <div className={styles['coz-inline-form']}>
          <input
            className={classNames(styles['coz-input-text'], this.state.hasError && styles['error'])}
            type='text'
            name='album-name'
            onInput={(event) => this.onNameChange(event)}
            disabled={this.state.isBusy}
            placeholder={t('Albums.create.inline_form.placeholder')}
            value={this.state.name}
            />
          <button
            className={classNames('coz-btn', 'coz-btn--regular', styles['coz-btn'])}
            disabled={this.state.isSubmitDisabled || this.state.isBusy}
            aria-busy={this.state.isBusy}
            >
            {t('Albums.create.inline_form.create_button')}
          </button>
        </div>
      </form>
    )
  }
}

export default translate()(CreateAlbumForm)
