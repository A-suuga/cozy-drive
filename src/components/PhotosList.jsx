import styles from '../styles/photoList'

import React from 'react'
import { translate } from '../lib/I18n'

import Empty from './Empty'
import Photo from './Photo'

import classNames from 'classnames'

export const PhotosList = ({ key, title, photos, selected, onPhotoToggle }) => {
  return (
    <div
      className={classNames(styles['pho-section'], selected.length && styles['pho-section--has-selection'])}
      key={key}
      >
      <h3>{title}</h3>
      {photos.map(photo =>
        <Photo
          photo={photo}
          key={photo._id}
          selected={selected.indexOf(photo._id) !== -1}
          onToggle={onPhotoToggle}
        />
      )}
      {photos.length === 0 && <Empty emptyType='photos' />}
    </div>
  )
}

export default translate()(PhotosList)
