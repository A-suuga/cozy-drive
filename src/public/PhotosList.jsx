import React from 'react'

import justifiedLayout from 'justified-layout'

const getStyle = box => ({
  width: `${box.width}px`,
  height: `${box.height}px`,
  top: `${box.top}px`,
  left: `${box.left}px`,
  position: 'absolute'
})

const toBox = photo => ({
  width: photo.attributes.metadata.width,
  height: photo.attributes.metadata.height
})

const PhotosList = ({photos}) => {
  const layout = justifiedLayout(photos.map(toBox))
  const boxes = layout.boxes.map((box, index) => (
    <img style={getStyle(box)} src={photos[index].url} />
  ))
  return <div style={{position: 'relative'}}>{boxes}</div>
}

export default PhotosList
