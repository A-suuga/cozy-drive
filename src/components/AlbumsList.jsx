import styles from '../styles/albumsList'

import React from 'react'

import Empty from '../components/Empty'
import AlbumItem from '../components/AlbumItem'

export const AlbumsList = ({ t, f, albums, onServerError }) => {
  if (albums.length === 0) {
    return <Empty emptyType='albums' />
  } else {
    return (
      <div className={styles['pho-album-list']}>
        {albums.map((a) => <AlbumItem album={a} key={a._id} onServerError={onServerError} />)}
      </div>
    )
  }
}

export default AlbumsList
