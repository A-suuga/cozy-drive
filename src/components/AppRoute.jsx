import React from 'react'
import { Route, Redirect } from 'react-router'

import Timeline from '../containers/Timeline'

import App from './App'
const ComingSoon = () => (<p style='margin-left: 2em'>Coming soon!</p>)

const AppRoute = (
  <Route component={App}>
    <Redirect from='/' to='photos' />
    <Route path='photos' component={Timeline} />
    <Route path='albums' component={ComingSoon} />
    <Route path='shared' component={ComingSoon} />
    <Route path='trash' component={ComingSoon} />
  </Route>
)

export default AppRoute
