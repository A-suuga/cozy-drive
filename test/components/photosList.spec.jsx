'use strict'

/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { mockT, mockF } from '../lib/I18n'
import { PhotosList } from '../../src/components/PhotosList'

const mockFetchedPhotos = [
  {
    _id: '33dda00f0eec15bc3b3c59a615001ac8',
    created_at: '0001-01-01T00:00:00Z',
    name: 'MonImage.jpg',
    size: '150000',
    updated_at: '0001-01-01T00:00:00Z'
  }
]

describe('PhotosList component', () => {
  it('should render correctly a timeline of photos according a photos array', () => {
    const component = renderer.create(
      <PhotosList t={mockT} f={mockF} photos={mockFetchedPhotos} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly a loading view if isIndexing is true', () => {
    const component = renderer.create(
      <PhotosList t={mockT} f={mockF} photos={mockFetchedPhotos} isIndexing />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly a loading view if isFetching is true', () => {
    const component = renderer.create(
      <PhotosList t={mockT} f={mockF} photos={mockFetchedPhotos} isFetching />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly a loading view if isWorking is true', () => {
    const component = renderer.create(
      <PhotosList t={mockT} f={mockF} photos={mockFetchedPhotos} isWorking />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly a loading view if isFirstFetch is true', () => {
    const component = renderer.create(
      <PhotosList t={mockT} f={mockF} photos={mockFetchedPhotos} isFirstFetch />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly an empty view if photos is empty', () => {
    const component = renderer.create(
      <PhotosList t={mockT} f={mockF} photos={[]} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
