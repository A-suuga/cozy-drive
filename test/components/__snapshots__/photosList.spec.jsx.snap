exports[`PhotosList component should render correctly a loading view if isFetching is true 1`] = `
<div
  class="pho-loading">
  <p>
    Loading
  </p>
</div>
`;

exports[`PhotosList component should render correctly a loading view if isFirstFetch is true 1`] = `
<div
  class="pho-loading">
  <p>
    Loading
  </p>
</div>
`;

exports[`PhotosList component should render correctly a loading view if isIndexing is true 1`] = `
<div
  class="pho-loading">
  <p>
    Your photos are currently being indexed. Please wait.
  </p>
</div>
`;

exports[`PhotosList component should render correctly a loading view if isWorking is true 1`] = `
<div
  class="pho-loading">
  <p>
    Uploading
  </p>
</div>
`;

exports[`PhotosList component should render correctly a timeline of photos according a photos array 1`] = `
<div
  role="contentinfo">
  <div
    class="pho-section">
    <h3>
      January 0001
    </h3>
    <img
      class="pho-photo"
      src="http://cozy.local:8080/files/download/33dda00f0eec15bc3b3c59a615001ac8" />
  </div>
</div>
`;

exports[`PhotosList component should render correctly an empty view if photos is empty 1`] = `
<div
  role="contentinfo">
  <div>
    <div
      class="pho-empty">
      <h2>
        You don\'t have any photos yet.
      </h2>
      <p>
        Click the \"Upload\" button to add photos.
      </p>
    </div>
  </div>
</div>
`;
